generic3g/tests/Test_CopyTransform.pf

```diff
@@ -0,0 +1,302 @@
+      character(len=*), intent(in) :: name
+      type(ESMF_TypeKind_Flag), intent(in) :: typekind
+      type(ESMF_StateItem_Flag), intent(in) :: itemtype
+      integer, optional, intent(out) :: rc
+      integer :: status
+      type(ESMF_Field) :: field
+      type(ESMF_FieldBundle) :: bundle
+      character(len=ESMF_MAXSTR) :: field_name
+      logical :: make_bundle
+
+      make_bundle = itemtype == MAPL_STATEITEM_FIELDBUNDLE
+      field_name = name
+      if(make_bundle) field_name = BUNDLE_FIELD_NAME
+      field = ESMF_FieldCreate(grid=grid, name=field_name, typekind=typekind, _RC)
+      if(typekind==ESMF_TYPEKIND_R4) then
+         call ESMF_FieldGet(field, farrayPtr=fptr_R4, _RC)
+         fptr_R4 = R4_VALUE
+      else
+         call ESMF_FieldGet(field, farrayPtr=fptr_R8, _RC)
+         fptr_R8 = R8_VALUE
+      end if
+      if(make_bundle) then 
+         bundle = ESMF_FieldBundleCreate(fieldList=[field], name=name, _RC)
+         call ESMF_StateAdd(state, fieldbundleList=[bundle], _RC)
+      else
+         call ESMF_StateAdd(state, fieldList=[field], _RC)
+      end if
+      _RETURN(_SUCCESS)
+
+   end subroutine initialize_state
+      
+   subroutine destroy_states(rc)
+      integer, optional, intent(out) :: rc
+      integer :: status
+
+      call destroy(importState, _RC)
+      call destroy(exportState, _RC)
+      _RETURN(_SUCCESS)
+
+   end subroutine destroy_states
+
+   subroutine destroy_state(state, rc)
+      type(ESMF_State), intent(inout) :: state
+      integer, optional, intent(out) :: rc
+      integer :: status
+      type(ESMF_StateItem_Flag), allocatable :: itemTypeList(:)
+      character(len=ESMF_MAXSTR), allocatable :: itemNameList(:)
+      character(len=ESMF_MAXSTR), allocatable :: names(:)
+      integer :: itemCount
+      type(ESMF_Field), allocatable :: fields(:)
+      type(ESMF_FieldBundle), allocatable :: bundles(:)
+
+      call ESMF_StateGet(state, itemCount=itemCount, _RC)
+      allocate(itemTypeList(itemCount))
+      allocate(itemNameList(itemCount))
+      call ESMF_StateGet(state, itemTypeList=itemTypeList, itemNameList=itemNameList, _RC)
+      names = get_item_names(itemNameList, itemTypeList, MAPL_STATEITEM_FIELD)
+      call get_fields(state, names, fields, _RC)
+      call destroy(fields, _RC)
+      names = get_item_names(itemNameList, itemTypeList, MAPL_STATEITEM_FIELDBUNDLE)
+      call get_bundles(state, names, bundles, _RC)
+      call destroy(bundles, _RC)
+      call ESMF_StateDestroy(state, _RC)
+      _RETURN(_SUCCESS)
+
+   end subroutine destroy_state
+
+   subroutine destroy_bundles(bundles, rc)
+      type(ESMF_FieldBundle), intent(inout) :: bundles(:)
+      integer, optional, intent(out) :: rc
+      integer :: status
+      integer :: i
+      
+      do i=1, size(bundles)
+         call destroy(bundles(i), _RC)
+      end do
+      _RETURN(_SUCCESS)
+     
+   end subroutine destroy_bundles
+
+   subroutine destroy_bundle(bundle, rc)
```

Comment:
The various destroy methods you have created in this test suite and the other test suite in this PR seem like they could be general utilities.

Immediate thought (but please consider alternatives) would be to have `MAPL_StateDestroy(state, recurse, rc)` where `recurse` is optional and false by default.      It is less clear to me the utility for the form that takes an array of bundles - seems an unusual pattern in practice, but could just be I've not worked in that corner of MAPL.  (Discuss with @bena-nasa )